plugins/modules/meraki_ms_access_policies.py

```diff
@@ -0,0 +1,583 @@
+                "suspendPortBounce": meraki.params["suspend_port_bounce"],
+            },
+            "failedAuthVlanId": None,
+            "reAuthenticationInterval": None,
+        },
+        "radiusCoaSupportEnabled": meraki.params["radius_coa_enabled"],
+        "hostMode": meraki.params["host_mode"],
+        "accessPolicyType": meraki.params["access_policy_type"],
+        "guestVlanId": meraki.params["guest_vlan"],
+        "voiceVlanClients": meraki.params["voice_vlan_clients"],
+        "urlRedirectWalledGardenEnabled": False,
+        "radiusAccountingEnabled": meraki.params["radius_accounting_enabled"],
+        "radiusAccountingServers": meraki.params["radius_accounting_servers"],
+        "systems_management_enrollment": meraki.params[
+            "systems_management_enrollment"
+        ],
+    }
+
+    if meraki.params["state"] == "query":
+        if meraki.params["number"]:
+            path = meraki.construct_path(
+                "get_one",
+                net_id=net_id,
+                custom={
+                    "number": meraki.params["number"],
+                },
+            )
+            response = meraki.request(path, method="GET")
+            meraki.result["data"] = response
+        else:
+            path = meraki.construct_path("get_all", net_id=net_id)
+            response = meraki.request(path, method="GET")
+            meraki.result["data"] = response
+    elif meraki.params["state"] == "present":
+
+        query_path_all = meraki.construct_path(
+            "get_all",
+            net_id=net_id,
+        )
+
+        original_all = meraki.request(query_path_all, method="GET")
+
+        for i in original_all:
+            if i.get("name") == meraki.params["name"]:
+                meraki.params["number"] = i.get("accessPolicyNumber")
+
+        if meraki.params["number"] is None:
+            path = meraki.construct_path(
+                "create",
+                net_id=net_id,
+            )
+            if meraki.params["auth_method"] == "Meraki authentication":
+                response = meraki.request(
+                    path, method="POST", payload=json.dumps(payload_auth)
+                )
+                meraki.result["data"] = response
+            elif meraki.params["auth_method"] == "my RADIUS server":
+                response = meraki.request(
+                    path, method="POST", payload=json.dumps(payload_radius)
+                )
+                meraki.result["data"] = response
+        else:
+            query_path = meraki.construct_path(
+                "get_one",
+                net_id=net_id,
+                custom={
+                    "number": meraki.params["number"],
+                },
+            )
+
+            update_path = meraki.construct_path(
+                "update",
+                net_id=net_id,
+                custom={
+                    "number": meraki.params["number"],
+                },
+            )
+
+            proposed = ""
```

Comment:
Assigning `proposed` to an empty string probably causes `is_update_required` to always be true so that code path executes. It should be comparing to the payload or whatever you're going to send. If I'm understanding the logic correctly, it would be either `payload_auth` or `payload_radius`.